src/plugins/discord/fetcher.js

```diff
@@ -0,0 +1,142 @@
+// @flow
+
+import * as Model from "./models";
+
+type FetchEndpoint = (endpoint: string) => Promise<any>;
+
+/**
+ * Provide the Guild ID to fetch against, and the 'limit'
+ * parameter when fetching GuildMembers, Messages, and Reactions.
+ *
+ */
+type FetchOptions = {|
+  guildId: Model.Snowflake,
+  membersLimit: number,
+  messagesLimit: number,
+  reactionsLimit: number,
+|};
+
+export type EndCursor = Model.Snowflake;
```

Comment:
Nit: I think the type alias doesn't make the function signatures easier to understand.
Let's just call them snowflakes?

Reply:
Updated

Reply:
Ah, I didn't mean to rename `Model.Snowflake` to `Snowflake`. I meant to remove the `EndCursor` type and use `(Model.)Snowflake` instead.